warehouse/transformer/logger.go

```diff
@@ -55,44 +48,45 @@ func (t *Transformer) CompareAndLog(
 	t.loggedEvents += int64(len(logEntries))
 }
 
-func (t *Transformer) differingEvents(
-	eventsToTransform []types.TransformerEvent,
-	pResponse, wResponse types.Response,
-	eventsByMessageID map[string]types.SingularEventWithReceivedAt,
-) ([]types.SingularEventT, string) {
+func (t *Transformer) differingEvents(eventsToTransform []types.TransformerEvent, pResponse, wResponse types.Response) ([]types.TransformerEvent, string) {
+	sortTransformerResponsesByJobID(pResponse.Events)
+	sortTransformerResponsesByJobID(pResponse.FailedEvents)
+	sortTransformerResponsesByJobID(wResponse.Events)
+	sortTransformerResponsesByJobID(wResponse.FailedEvents)
+
 	// If the event counts differ, return all events in the transformation
 	if len(pResponse.Events) != len(wResponse.Events) || len(pResponse.FailedEvents) != len(wResponse.FailedEvents) {
-		events := lo.Map(eventsToTransform, func(e types.TransformerEvent, _ int) types.SingularEventT {
-			return eventsByMessageID[e.Metadata.MessageID].SingularEvent
-		})
-		t.stats.mismatchedEvents.Observe(float64(len(events)))
-		return events, ""
+		t.stats.mismatchedEvents.Observe(float64(len(eventsToTransform)))
+		return eventsToTransform, ""
 	}
 
 	var (
-		differedSampleEvents []types.SingularEventT
-		differedEventsCount  int
-		sampleDiff           string
+		differedEventsCount int
+		sampleDiff          string
 	)
 
 	for i := range pResponse.Events {
 		diff := cmp.Diff(wResponse.Events[i], pResponse.Events[i])
 		if len(diff) == 0 {
 			continue
 		}
-
 		if differedEventsCount == 0 {
-			// Collect the mismatched messages and break (sample only)
-			differedSampleEvents = append(differedSampleEvents, lo.Map(pResponse.Events[i].Metadata.GetMessagesIDs(), func(msgID string, _ int) types.SingularEventT {
-				return eventsByMessageID[msgID].SingularEvent
-			})...)
 			sampleDiff = diff
 		}
 		differedEventsCount++
 	}
 	t.stats.matchedEvents.Observe(float64(len(pResponse.Events) - differedEventsCount))
 	t.stats.mismatchedEvents.Observe(float64(differedEventsCount))
-	return differedSampleEvents, sampleDiff
+	if differedEventsCount == 0 {
+		return nil, ""
+	}
+	return eventsToTransform, sampleDiff
```

Comment:
Is `eventsToTransform` return value same as the input argument? If yes, why do we need to return it?

Reply:
Done.